Makefile

```diff
@@ -229,39 +234,98 @@ build-all-in-one-debug build-agent-debug build-query-debug build-collector-debug
+	$(GOBUILD) $(DISABLE_OPTIMIZATIONS) -tags ui -o ./cmd/query/query$(SUFFIX)-$(GOOS)-$(GOARCH) $(BUILD_INFO) ./cmd/query/
 
 .PHONY: build-collector build-collector-debug
 build-collector build-collector-debug:
-	$(GOBUILD) $(DISABLE_OPTIMIZATIONS) -o ./cmd/collector/collector$(SUFFIX)-$(GOOS)-$(GOARCH) $(BUILD_INFO) ./cmd/collector/main.go
+	$(GOBUILD) $(DISABLE_OPTIMIZATIONS) -o ./cmd/collector/collector$(SUFFIX)-$(GOOS)-$(GOARCH) $(BUILD_INFO) ./cmd/collector/
 
 .PHONY: build-ingester build-ingester-debug
 build-ingester build-ingester-debug:
-	$(GOBUILD) $(DISABLE_OPTIMIZATIONS) -o ./cmd/ingester/ingester$(SUFFIX)-$(GOOS)-$(GOARCH) $(BUILD_INFO) ./cmd/ingester/main.go
+	$(GOBUILD) $(DISABLE_OPTIMIZATIONS) -o ./cmd/ingester/ingester$(SUFFIX)-$(GOOS)-$(GOARCH) $(BUILD_INFO) ./cmd/ingester/
 
 .PHONY: build-remote-storage build-remote-storage-debug
 build-remote-storage build-remote-storage-debug:
-	$(GOBUILD) $(DISABLE_OPTIMIZATIONS) -o ./cmd/remote-storage/remote-storage$(SUFFIX)-$(GOOS)-$(GOARCH) $(BUILD_INFO) ./cmd/remote-storage/main.go
+	$(GOBUILD) $(DISABLE_OPTIMIZATIONS) -o ./cmd/remote-storage/remote-storage$(SUFFIX)-$(GOOS)-$(GOARCH) $(BUILD_INFO) ./cmd/remote-storage/
 
 .PHONY: build-binaries-linux
 build-binaries-linux:
 	GOOS=linux GOARCH=amd64 $(MAKE) build-platform-binaries
 
+# Magic values:
+# - LangID "0409" is "US-English".
+# - CharsetID "04B0" tranlsates to decimal 1200 for "Unicode".
+# - FileOS "040004" defines the Windows kernel "Windows NT".
+# - FileFlagMask, FileFlags, FileOS, FileType, FileSubType are values taken from the example.
+# - FileType "01" is "Application".
+define VERSIONINFO
+{
+    "FixedFileInfo": {
+        "FileVersion": {
+            "Major": $(GIT_CLOSEST_TAG_MAJOR),
+            "Minor": $(GIT_CLOSEST_TAG_MINOR),
+            "Patch": $(GIT_CLOSEST_TAG_PATCH),
+            "Build": 0
+        },
+        "ProductVersion": {
+            "Major": $(GIT_CLOSEST_TAG_MAJOR),
+            "Minor": $(GIT_CLOSEST_TAG_MINOR),
+            "Patch": $(GIT_CLOSEST_TAG_PATCH),
+            "Build": 0
+        },
+        "FileFlagsMask": "3f",
+        "FileFlags ": "00",
+        "FileOS": "040004",
+        "FileType": "01",
+        "FileSubType": "00"
+    },
+    "StringFileInfo": {
+        "FileDescription": "NAME",
+        "FileVersion": "$(GIT_CLOSEST_TAG_MAJOR).$(GIT_CLOSEST_TAG_MINOR).$(GIT_CLOSEST_TAG_PATCH).0",
+        "LegalCopyright": "2015-2023 The Jaeger Project Authors",
+		"ProductName": "NAME",
+        "ProductVersion": "$(GIT_CLOSEST_TAG_MAJOR).$(GIT_CLOSEST_TAG_MINOR).$(GIT_CLOSEST_TAG_PATCH).0"
+    },
+    "VarFileInfo": {
+        "Translation": {
+            "LangID": "0409",
+            "CharsetID": "04B0"
+        }
+    }
+}
+endef
+
+export VERSIONINFO
+
+.PHONY: prepare-winres
+prepare-winres:
+	echo $$VERSIONINFO | sed "s/NAME/Jaeger Agent/g" | $(GOVERSIONINFO) -o="cmd/agent/$(SYSOFILE)" -
```

Comment:
I think you can use `define VERSIONINFO` as a function:
```
echo $(call VERSIONINFO,Jaeger Agent) | ...
```
and access the parameter as `$1`

You can also define another helper target to reduce boiler plate, e.g. reduce each command to 
```
  $(MAKE) prepare-winres-once NAME="Jaeger Agent" PATH="cmd/agent"
```
(or use another template / function for that)